cardano-db-sync/src/Cardano/DbSync/OffChain/Http.hs

```diff
@@ -78,13 +79,33 @@ httpGetOffChainPoolData manager request purl expectedMetaHash = do
     url = OffChainPoolUrl purl
 
 httpGetOffChainVoteData ::
-  Http.Manager ->
-  Http.Request ->
+  [Text] ->
+  VoteUrl ->
+  Maybe VoteMetaHash ->
+  DB.AnchorType ->
+  ExceptT OffChainFetchError IO SimplifiedOffChainVoteData
+httpGetOffChainVoteData gateways vurl metaHash anchorType = do
+  case useIpfsGatewayMaybe vurl gateways of
+    Nothing -> httpGetOffChainVoteDataSingle vurl metaHash anchorType
+    Just [] -> left $ OCFErrNoIpfsGateway (OffChainVoteUrl vurl)
```

Comment:
Why not just make this a `NonEmpty`?

Reply:
It can be empty

Reply:
I might be reading this wrong, but it looks to me like having empty `gateways` is an error here. If that's correct, can we make gateways a nonempty? That would push the check up to a config check and make everywhere else a safer call

Reply:
It's an error that doesn't bubble up. It ends up in the off_chain_vote_fetch_error table.
The user has to specify a list of gateaways, if he wants to support ipfs links